### PR TITLE
feat: store svm recent blockhashes

### PR DESCRIPTION
**Pull request by guibescos**

### CODE REVIEW

File: auction-server/src/server.rs [hunk 1]
```diff
@@ -93,11 +95,10 @@ use {
 async fn fault_tolerant_handler<F, Fut>(name: String, f: F)
 where
     F: Fn() -> Fut,
-    Fut: Future<Output = anyhow::Result<()>> + Send + 'static,
-    Fut::Output: Send + 'static,
+    Fut: Future<Output = anyhow::Result<()>>,
 {
     loop {
-        let res = tokio::spawn(f()).await;
+        let res = AssertUnwindSafe(f()).catch_unwind().await;
```
Comment (guibescos) [inline auction-server/src/server.rs:101]: This is a superior way of doing this; there's no reason to spawn a task that we await immediately.
Internally tokio uses AssertUnwindSafe too.

File: auction-server/src/api.rs [hunk 2]
```diff
@@ -404,7 +404,7 @@ pub async fn start_api(run_options: RunOptions, store: Arc<StoreNew>) -> Result<
 
     // The generated OpenAPI document contains "discriminator" fields which are not generated correctly to be supported by redoc
     // We need to remove them from the document to make sure redoc can render the document correctly
-    let original_doc = serde_json::to_value(&ApiDoc::openapi())
+    let original_doc = serde_json::to_value(ApiDoc::openapi())
```
Comment (guibescos) [inline auction-server/src/api.rs:407]: clippy did this

File: auction-server/src/auction.rs [hunk 3]
```diff
@@ -1340,6 +1355,21 @@ impl ChainStore for ChainStoreSvm {
     const CHAIN_TYPE: models::ChainType = models::ChainType::Svm;
     const AUCTION_MINIMUM_LIFETIME: Duration = Duration::from_millis(400);
 
+    async fn get_next_trigger<'a>(
```
Comment (danimhr) [inline auction-server/src/auction.rs:1358]: is it possible to extract slot number from this block trigger?
We need to know slot number to make sure running the conclusion loop for svm every 150 slots.

File: auction-server/src/auction.rs [hunk 4]
```diff
@@ -1348,8 +1378,19 @@ impl ChainStore for ChainStoreSvm {
...
-        Ok(slot_subscribe)
+        let (block_subscribe, _) = client
+            .block_subscribe(
+                solana_client::rpc_config::RpcBlockSubscribeFilter::All,
+                Some(RpcBlockSubscribeConfig {
+                    encoding:                          None,
+                    transaction_details:               Some(TransactionDetails::None),
+                    show_rewards:                      Some(false),
+                    max_supported_transaction_version: None,
+                    commitment:                        Some(CommitmentConfig::confirmed()),
```
Comment (danimhr) [inline auction-server/src/auction.rs:1389]: is it possible to listen for `finalized` blocks?
we want to sign the transaction with this block hash and it's safer to use `finalized` for this

File: auction-server/src/server.rs (hunk 1 above)
Comment (danimhr) [reply auction-server/src/server.rs:101]: Could you please describe what's the difference btw these two methods?

File: auction-server/src/server.rs [hunk 5]
```diff
@@ -380,6 +381,7 @@ fn setup_svm(
             (
                 chain_id.clone(),
                 ChainStoreSvm {
+                    recent_blockhash: None.into(),
```
Comment (danimhr) [inline auction-server/src/server.rs:384]: We can store the initial block hash by creating a single RPC request when setting up the server. It's better to not let this value be None

File: auction-server/Cargo.toml [hunk 6]
```diff
@@ -49,6 +49,7 @@ serde_with = { version = "3.9.0", features = ["base64"] }
 anchor-lang = "0.30.1"
 express-relay = { path = "../contracts/svm/programs/express_relay" }
 solana-rpc-client = "2.0.13"
+solana-transaction-status = "2.0.13"
```
Comment (danimhr) [inline auction-server/Cargo.toml:52]: Do we need it?

File: auction-server/src/api.rs (hunk 2 above)
Comment (danimhr) [inline auction-server/src/api.rs:407]: :-? what's the problem with the previous version?

**Pull request closed without merging**